### ISSUE TITLE
Checking macro prints not-falsey-or-exception? when fails

### ISSUE THREAD

**Issue by firesofmay** [gfredericks/test.chuck#17]
Hi,

I was trying out checking macro and I found the output a little weird.
Here's a sample test:

```
(deftest do-fail
  (checking "fail me" 1000
            [n gen/nat]
            (is (not= 10 n))))
```

And here's the output:
```
> (do-fail)

FAIL in (do-fail) (clojure_test.clj:17)
fail me
{:result false, :seed 1439064390947, :failing-size 48, :num-tests 49, :fail [10], :shrunk {:total-nodes-visited 4, :depth 0, :result false, :smallest [10]}}
expected: (not-falsey-or-exception? (:result result))
  actual: (not (not-falsey-or-exception? false))

FAIL in (do-fail) (demo.clj:32)
fail me
expected: (not= 10 n)
  actual: (not (not= 10 10))
```

Why does it print this?
```
expected: (not-falsey-or-exception? (:result result))
actual: (not (not-falsey-or-exception? false))
```

**Comment by lackita** [comment]
Looking into this now.

**Comment by gfredericks** [comment]
hey @lackita would the implementation for `checking` be easier if `test.check` provided callbacks for all the different events of the test-running process?

**Comment by lackita** [comment]
@gfredericks 
Maybe a little bit, the code currently assumes that the reports from the final failure are the smallest example, but I think I remember noticing minor discrepancies in practice. I might be able to eliminate those if I had a callbacks around successes, failures, and done shrinking.

**Comment by gfredericks** [comment]
Okay cool; I was working on such a thing at one point, so might be able to get it into the `0.9.0` release.

**Comment by lackita** [comment]
Alright, the specific trigger I'd find useful is one fired when a smaller
example of a failure is identified.

**Comment by gfredericks** [comment]
I'm curious if that's because you would print something on each smaller failure?

**Comment by gfredericks** [comment]
I'm wondering because I've done such a thing myself in a test.check fork, and it ends up being very verbose, which makes me want to have a way to control verbosity.

**Comment by lackita** [comment]
Actually, quite the opposite. I only want to report the smallest failure
and I need a way of determining if this new failure is the new smallest.

**Comment by gfredericks** [comment]
why can't you do that just by waiting for the whole `quick-check` call to return and just examining the `:shrunk` key?

**Comment by lackita** [comment]
I wanted to present the actual failure report from those values, not the
value leading to the failure.

**Comment by gfredericks** [comment]
oh right, and you need to distinguish it from the subsequent hundreds of test runs that passed before the test runner gave up. Got it.

**Comment by firesofmay** [comment]
@lackita Any updates on this one?

**Comment by lackita** [comment]
Sorry, I've been on vacation and haven't had much time to investigate.
That's over tomorrow so I'll be looking at it some more on the commute
home.

**Comment by firesofmay** [comment]
Sure, no problem :)

**Comment by lackita** [comment]
It looks like what happened is the spurious line was necessary when it was written, but then I later started capturing and propagating reports in another way and it became redundant.

**Comment by firesofmay** [comment]
@lackita Thanks will check it out. :)
@gfredericks Any idea when are we releasing the next version?

**Comment by gfredericks** [comment]
Apparently this created a new issue (#23) so I will probably release when that gets figured out

**Comment by gfredericks** [comment]
Reopening this since it caused test failures.

**Comment by firesofmay** [comment]
@lackita Any updates on this?

**Comment by lackita** [comment]
Sorry, I'll try to take a look on the commute home today.

**Comment by firesofmay** [comment]
@lackita Hey any update on this?

**Comment by lackita** [comment]
@firesofmay Sorry, this got conflated with a conversations in #25, I believe the issue should be fixed now.

**Comment by firesofmay** [comment]
No problem. Thanks for all your help :)

**Comment by gfredericks** [comment]
@firesofmay if you can confirm this fixes the problem I can make a release for it

**Comment by firesofmay** [comment]
@gfredericks LGTM! :+1:

**Comment by gfredericks** [comment]
Okay, this is released with `0.1.22`
